src/System.Numerics.Vectors/tests/CommonHashCodeTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Numerics.Tests
+{
+    public static class CommonHashCodeTests
+    {
+        private const int ElementsCount = 100000;
+
+        private static void CountVector2Collisions(Func<int, Vector2> vectorConstructor, out int collisions, out int maxCollisions)
+        {
+            var coll = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Select(x => new
+                {
+                    Hash = x.Key,
+                    x.ToList().Count
+                }).ToList();
+
+            int collidedHashesCount = coll.Count(x => x.Count > 1);
+
+            collisions = coll.Where(x => x.Count > 1).Sum(x => x.Count) - collidedHashesCount;
+
+            maxCollisions = coll.Max(x => x.Count);
+        }
+
+        private static void CountVectorCollisions<T>(Func<int, Vector<T>> vectorConstructor,
+            out int collisions) where T : struct
+        {
+            var coll = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Select(x => new
+                {
+                    Hash = x.Key,
+                    x.ToList().Count
+                }).ToList();
+
+            int collidedHashesCount = coll.Count(x => x.Count > 1);
+
+            collisions = coll.Where(x => x.Count > 1).Sum(x => x.Count) - collidedHashesCount;
+        }
+
+        [Fact]
+        public static void CombineHashCodesVector2Tests()
+        {
+            int collisions;
+            int maxCollisions;
+
+            // On Highly distrubuted data, percent of collisions has to be less than 1%.
+            CountVector2Collisions(x => new Vector2 {X = x, Y = x + ElementsCount}, out collisions, out maxCollisions);
+            Assert.True(0.01 >= collisions/(float) ElementsCount);
+
+            // Every coordinate not greater than 10, total collisions could be 99.9%, but maximum elements with equal hash < 25%.
+            CountVector2Collisions(x => new Vector2 {X = x%10, Y = (x + ElementsCount)%10}, out collisions, out maxCollisions);
```

Comment:
NO.
This test is still worthless.  You're still only iterating over 10 elements, which is a miniscule portion of the entire range.  How small?  **0.0000000046566**  They could all collide, but the overall percentage still be way below the earlier range you check.  (Although all 10 elements colliding with each other is probably still bad for distribution reasons, this is _not_ how I would check for such a thing.)
You need to go back and revisit my earlier comment here; the entire test needs to be redesigned.  For one thing, even though you're iterating over 100,000 elements, that represents only 0.0000465661288 of the entire range.  **You're examining fewer elements than your collision percentage**, by a factor of about 2000.  You need to at least adjust the percentage of collisions by the range you actually check, and a greater percentage of the elements (preferably, greater than the collision target range) would be better.